fusionSuite/frontend
# new: Manage users

**This PR must be merged after https://github.com/fusionSuite/frontend/pull/31**

Changes proposed in this pull request:

- Add a form to create users (first and last names, username and organization), username is prefilled with first and last names
- Add a page to list and delete users
- Update the French locale
- On a technical aspect, it introduces "models" to manipulate items more easily

How to test the feature manually:

1. Login, go to the "users" page and click on "Create a user"
2. Create a bunch of users
3. Verify that the username is prefilled based on first and last names
4. Go back to the users list and check that users have been created

Known limitations:

- considering the current state of the backend:
  - usernames can be duplicated, and there are no verification of the format (e.g. spaces are allowed)
  - emails and passwords cannot be set yet
  - first and last names are optional
- some code could be refactored with the organizations, but I plan to do it after

List of users:

![Capture d’écran 2022-08-30 à 10 52 00](https://user-images.githubusercontent.com/107053378/187393813-127d7049-4e52-4603-a0bf-4125f79ecf54.png)

Create a user:

![Capture d’écran 2022-08-30 à 10 52 18](https://user-images.githubusercontent.com/107053378/187393859-35813d34-7a79-44b1-95d4-50cd0bcf0327.png)

Pull request checklist:


_If you think one of the item isn’t applicable to the PR, please check it
anyway and precise `N/A` next to it._

Related to https://github.com/fusionSuite/FusionSuite/issues/42
